src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -0,0 +1,131 @@
+  values: string[];
+  label: string;
+  allowMultiple: boolean;
+  disabled?: boolean;
+  onChange: (values: string[]) => void;
+};
+
+const CheckboxWithTextfields = (props: CheckboxWithTextfieldsProps) => {
+  const [isChecked, setIsChecked] = useState(false);
+  const [values, setValues] = useState<(string | null)[]>(props.values);
+
+  const [shouldShowAddButton, setShowAddButton] = useState(
+    Boolean(values[values.length - 1])
+  );
+
+  useEffect(() => {
+    if (!props.values.length) {
+      setValues([null]);
+    } else {
+      setValues(props.values);
+    }
+
+    setIsChecked(props.values.length > 0);
+    setShowAddButton(Boolean(props.values[props.values.length - 1]));
+  }, [props.values]);
+
+  const handleCheckboxOnChange = (isChecked: boolean) => {
+    setIsChecked(isChecked);
+    if (!isChecked && values.length > 0) {
+      props.onChange([]);
+    }
+  };
+
+  const addEntry = () => {
+    setShowAddButton(false);
+    setValues([...values, null]);
+  };
+
+  const handleOnChange = (value: string, index: number) => {
+    const newValues: string[] = [...values] as string[];
+    newValues[index] = value;
+
+    setShowAddButton(Boolean(newValues[newValues.length - 1]));
+
+    props.onChange(newValues);
+  };
+
+  const handleOnRemove = (index: number) => {
+    const newValues: string[] = [...values] as string[];
+
+    newValues.splice(index, 1);
+    setShowAddButton(Boolean(newValues[newValues.length - 1]));
+
+    props.onChange(newValues);
+  };
+
+  const valuesWithoutFirst: string[] = [...values] as string[];
+  valuesWithoutFirst.shift();
+
+  return (
+    <table className={styles.wrapperTable}>
+      <tbody>
+        <tr>
+          <td className={styles.checkboxWrapper}>
+            <Checkbox
+              checked={isChecked}
+              onChange={handleCheckboxOnChange}
+              label={props.label}
+              disabled={props.disabled}
+            />
+          </td>
+          <td>
+            <div className={styles.fieldsWrapper}>
+              {isChecked && (
+                <div key={0}>
+                  <PasteOrUpload
+                    value={values[0]}
+                    onChange={(value) => handleOnChange(value, 0)}
+                    onRemove={() => handleOnRemove(0)}
+                    placeholder={'Paste data'}
+                  />
+                </div>
+              )}
+
+              {isChecked &&
+                valuesWithoutFirst.map((value: string, key: number) => {
+                  return (
+                    <div key={key + 1}>
+                      <PasteOrUpload
+                        value={value}
+                        onChange={(newValue) =>
+                          handleOnChange(newValue, key + 1)
+                        }
+                        onRemove={() => handleOnRemove(key + 1)}
+                        placeholder={'Paste data'}
+                      />
+                    </div>
+                  );
+                })}
```

Comment:
What is the difference in treatment of first value as opposed to valuesWithoutFirst?

Reply:
Related to the comment above. This logic has been updated.